runtime/calamari/tests/xcm_tests.rs

```diff
@@ -0,0 +1,1791 @@
+	ParaA::execute_with(|| {
+		assert_ok!(parachain::XTokens::transfer(
+			parachain::Origin::signed(ALICE.into()),
+			parachain::CurrencyId::MantaCurrency(a_currency_id),
+			amount,
+			Box::new(VersionedMultiLocation::V1(dest)),
+			dest_weight,
+		));
+		assert_eq!(
+			parachain::Balances::free_balance(&ALICE.into()),
+			INITIAL_BALANCE - amount
+		)
+	});
+
+	// Alice on B should receive nothing since we didn't specify the unit per second
+	ParaB::execute_with(|| {
+		assert_eq!(parachain::Assets::balance(a_currency_id, &ALICE.into()), 0);
+	});
+}
+
+#[test]
+fn receive_asset_with_is_sufficient_false(){
+	MockNet::reset();
+
+	let new_account = [5u8; 32];
+	let relay_asset_id = 0u32;
+	let source_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::parent()));
+	let asset_metadata = parachain::AssetRegistarMetadata {
+		name: b"Kusama".to_vec(),
+		symbol: b"KSM".to_vec(),
+		decimals: 12,
+		min_balance: 1u128,
+		evm_address: None,
+		is_frozen: false,
+		is_sufficient: false,
+	};
+	let amount = 123u128;
+
+	// register relay asset in parachain A
+	ParaA::execute_with(|| {
+		assert_ok!(parachain::AssetManager::register_asset(
+			parachain::Origin::root(),
+			source_location,
+			asset_metadata
+		));
+		// we don't charge anything during test
+		assert_ok!(parachain::AssetManager::set_units_per_second(
+			parachain::Origin::root(), 
+			relay_asset_id, 
+			0u128));
+	});
+
+	let dest: MultiLocation = AccountId32 {
+		network: Any,
+		id: new_account.into()
+	}.into();
+
+	Relay::execute_with(|| {
+		assert_ok!(RelayChainPalletXcm::reserve_transfer_assets(
+			relay_chain::Origin::signed(ALICE),
+			Box::new(X1(Parachain(1)).into().into()),
+			Box::new(VersionedMultiLocation::V1(dest.clone()).clone().into()),
+			Box::new((Here, amount).into()),
+			0,
+		));
+		assert_eq!(
+			relay_chain::Balances::free_balance(&para_account_id(1)),
+			INITIAL_BALANCE + amount
+		);
+	});
+
+	// parachain should not have received assets
+	ParaA::execute_with(|| {
+		assert_eq!(parachain::Assets::balance(relay_asset_id, &new_account.into()), 0);
+	});
+
+	// Send native token to fresh_account
+	ParaA::execute_with(|| {
+		assert_ok!(parachain::Balances::transfer(
+			parachain::Origin::signed(ALICE.into()),
+			new_account.into(),
+			100
+		));
+	});
+
+	Relay::execute_with(|| {
+		assert_ok!(RelayChainPalletXcm::reserve_transfer_assets(
+			relay_chain::Origin::signed(ALICE),
+			Box::new(X1(Parachain(1)).into().into()),
+			Box::new(VersionedMultiLocation::V1(dest).clone().into()),
+			Box::new((Here, amount).into()),
+			0,
+		));
+		assert_eq!(
+			relay_chain::Balances::free_balance(&para_account_id(1)),
+			INITIAL_BALANCE + amount + amount
+		);
+	});
+
+	// parachain should not have received assets
+	ParaA::execute_with(|| {
+		println!("fresh account bal: {}", parachain::Assets::balance(relay_asset_id, &new_account.into()));
+	});
+
+}
+
+#[test]
+fn receive_asset_with_is_sufficient_true(){
```

Comment:
Isn't this pretty much the same as `reserve_transfer_relaychain_to_parachain_a`

Reply:
yes, this is just a dual test to previous test to make sure under the exact same condition the test passes when `is_sufficient` is `true`.